components/MintPage/MintDay/MintBoard/MintBoard.tsx

```diff
@@ -1,71 +1,45 @@
 import { useState, useMemo, useEffect } from "react"
-import { useAccount } from "wagmi"
 import { useMeasure } from "react-use"
 import { useMediaQuery } from "usehooks-ts"
-import SectionContainer from "../SectionContainer"
-import Title from "../../Common/Title"
-import Content from "../../Common/Content"
-import Character from "../Character"
-import Media from "../../../shared/Media"
-import QuantityCard from "./QuantityCard"
-import { Button } from "../../../shared/Button"
-import WalletConnectButton from "../../WalletConnectButton"
-import ModalSelector from "./Modals/ModalSelector"
-import { useMintProvider } from "../../../providers/MintProvider"
+import SectionContainer from "../../SectionContainer"
+import Title from "../../../Common/Title"
+import Content from "../../../Common/Content"
+import Character from "../../Character"
+import Media from "../../../../shared/Media"
+import QuantityCard from "../QuantityCard"
+import ModalSelector from "../Modals/ModalSelector"
+import { useMintProvider } from "../../../../providers/MintProvider"
+import MintBoardButtons from "./ActionButtons"
 
 const MintBoard = () => {
-  const { hasPassport, hasNotFreeMintClaimed, hasFriendAndFamily } = useMintProvider()
+  const {
+    hasPassport,
+    hasUnclaimedFreeMint,
+    hasFriendAndFamily,
+    addToCart,
+    removeFromCart,
+    getCartTier,
+  } = useMintProvider()
   const [openModal, setOpenModal] = useState(false)
 
   const [boardRef, { height }] = useMeasure()
   const isXs = useMediaQuery("max-width: 393px")
 
-  const { isConnected } = useAccount()
-
-  const [tierIQuantity, setTierIQuantity] = useState(0)
-  const [tierIIQuantity, setTierIIQuantity] = useState(0)
-  const [tierIIIQuantity, setTierIIIQuantity] = useState(0)
-
   const automaticOpenModal = useMemo(
-    () => (hasPassport && hasNotFreeMintClaimed) || hasFriendAndFamily,
-    [hasPassport, hasNotFreeMintClaimed, hasFriendAndFamily],
+    () => (hasPassport && hasUnclaimedFreeMint) || hasFriendAndFamily,
+    [hasPassport, hasUnclaimedFreeMint, hasFriendAndFamily],
   )
 
   useEffect(() => {
     setOpenModal(automaticOpenModal)
   }, [automaticOpenModal])
 
   const increaseQuantity = (type: number) => {
-    switch (type) {
-      case 1:
-        setTierIQuantity(tierIQuantity + 1)
-        break
-      case 2:
-        setTierIIQuantity(tierIIQuantity + 1)
-        break
-      case 3:
-        setTierIIIQuantity(tierIIIQuantity + 1)
-        break
-      default:
-    }
+    addToCart(type)
   }
 
   const decreaseQuantity = (type: number) => {
-    switch (type) {
-      case 1:
-        if (!tierIQuantity) return
-        setTierIQuantity(tierIQuantity - 1)
-        break
-      case 2:
-        if (!tierIIQuantity) return
-        setTierIIQuantity(tierIIQuantity - 1)
-        break
-      case 3:
-        if (!tierIIIQuantity) return
-        setTierIIIQuantity(tierIIIQuantity - 1)
-        break
-      default:
-    }
```

Comment:
interesting. I went with a different way initially. I left this code as is and created an updateCart function that sets the carts tiers and quantity on button click.